internal/service/worker.go

```diff
@@ -459,16 +465,12 @@ func (w *Worker) processAnnotations(
 		}
 	}()
 
-	logger, err := w.TraceExtractor(ctx, w.Logger)
-	if err != nil {
-		return "", nil, fmt.Errorf("failed to get a logger with datadog link: %w", err)
-	}
-	logger.Debug().Any("annotations", annotations).Msg("Adding annotations to the PDF")
-
 	for _, annotation := range annotations {
 		var err error
+		var annSpan ddtrace.Span
 		switch v := annotation.(type) {
 		case domain.AnnotationCheckbox:
+			annSpan, _ = ddTracer.StartSpanFromContext(ctx, "PdfHandler.AddCheckboxToPage")
```

Comment:
Change a little the logic, to not start the span until you're sure you're on the correct page, otherwise you'll create a ton of spans that represent nothing. Add near the `ph.` calls.